src/robotFactory.js

```diff
@@ -1,10 +1,98 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
 
-class DeliveryDrone {}
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this.y;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this.y;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this.x;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this.x;
+  };
+
+  getInfo() {
+    // eslint-disable-next-line max-len
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, { x = 0, y = 0, z = 0 }, chipVersion) {
+    super(name, weight, {
+      x, y,
+    }, chipVersion);
+
+    this.coords = {
+      x,
+      y,
+      z,
+    };
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+
+    return this.z;
+  };
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+
+    return this.z;
+  };
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name,
+    weight,
+    { x = 0, y = 0, z = 0 },
+    chipVersion,
+    maxLoadWeight,
+    currentLoad = null) {
+    super(name, weight, chipVersion, {
+      x, y, z,
+    });
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
```

Comment:
Fix code style pls, i can't read this code

```suggestion
  constructor(
    name,
    weight,
    { x = 0, y = 0, z = 0 },
    chipVersion,
    maxLoadWeight,
    currentLoad = null
  ) {
    super(
      name,
      weight,
      chipVersion,
      { x, y, z }
    );

    this.maxLoadWeight = maxLoadWeight;
    this.currentLoad = currentLoad;
  }
```

Reply:
Лінтер не пропускав, тому пофіксила через disable-next-line

Reply:
No, let's move all args to new lines